### PR TITLE
Add User store

### PR DESCRIPTION
**Pull request by Tugduoff**
## Changes made

Added a new store in pinia for the user auth. It will be used to know which user is connected at anytime.

### CODE REVIEW

**Review by mathieu-MZ — APPROVED** [review]

**Review by rchevallier1234 — APPROVED** [review]

File: frontend/src/views/DashboardPage.vue [hunk 1]
```diff
@@ -104,7 +108,44 @@ onMounted(async () => {
...
+    if (!email && !token) {
+        console.error('Not logged in.');
+        router.push('/');
+        return;
+    } else {
+        const user = userStore.user;
+        console.log(user);
+        if (!user) {
+            const res: { status: number, data: User } = await axios.get<User>(
+                `http://localhost:8080/api/user`,
```
Comment (AdamLesage) [inline frontend/src/views/DashboardPage.vue:128]: Put deployed URL instead of locahost in env file

File: frontend/src/views/LoginPage.vue [hunk 2]
```diff
@@ -2,31 +2,48 @@
...
+
+const store = useUserStore();
 const hover = ref(false);
 
 // Form submission handler
 const handleSubmit = async (values: LoginFormValues) => {
     console.log('Login Form Received:', values);
     try {
-        const res = await axios.post('http://localhost:8080/auth/login', {
+        const res: { status: number, data: { user: User } } = await axios.post('http://localhost:8080/auth/login', {
```
Comment (AdamLesage) [inline frontend/src/views/LoginPage.vue:22]: change URL to the deployed one

File: frontend/src/views/SignUpPage.vue [hunk 3]
```diff
@@ -5,16 +5,21 @@ import LoginButton from '@/components/LoginButton.vue';
...
+
+const store = useUserStore();
 const hover = ref(false);
 
 // Form submission handler
 const handleSubmit = async (values: SignUpFormValues) => {
     console.log('Sign Up Form Received:', values);
     try {
-        const res: { status: number, data: { authToken: string } } = await axios.post<User>('http://localhost:8080/auth/register', {
+        const res: { status: number, data: User } = await axios.post<User>('http://localhost:8080/auth/register', {
```
Comment (AdamLesage) [inline frontend/src/views/SignUpPage.vue:22]: use deployed URL

File: frontend/src/views/UserDetailsPage.vue [hunk 4]
```diff
@@ -2,10 +2,50 @@
...
-const handleSubmit = (values: DetailsFormValues) => {
+const handleSubmit = async (values: DetailsFormValues) => {
     console.log('Login Form Received:', values);
+    if (!user) {
+        console.error('Not logged in.');
+        return;
+    }
+    try {
+        const authToken = Cookies.get('token');
+        const res: { status: number, data: User } = await axios.put<User>(`http://localhost:8080/api/user/${user.uuid}`, {
```
Comment (AdamLesage) [inline frontend/src/views/UserDetailsPage.vue:29]: use deployed URL

**Pull request merged**